src/dbnode/storage/bootstrap/process.go

```diff
@@ -302,5 +357,30 @@ func (b bootstrapProcess) newRunOptions() RunOptions {
 	return NewRunOptions().
 		SetCacheSeriesMetadata(
 			b.processOpts.CacheSeriesMetadata(),
-		)
+		).
+		SetInitialTopologyState(b.initialTopologyState)
+}
+
+// TopologyState represents a snapshot of the state of the topology at a
+// given moment.
+type TopologyState struct {
+	Self             string
```

Comment:
could you use a `topology.Host` for self too? and would Origin be more consistent a name?